### PR TITLE
feat(prometheus): add ignore_timestamp option

### LINKED ISSUE #9739: Add `ignore_timestamp` option to prometheus inputs plugin
### Proposal:
When some exporter exposes metrics with timestamp, prometheus input plugin uses the timestamp obtained as the timestamp for the metrics. However, in some situations, we want to use the timestamp when the metrics were obtained from the exporter. We need the ability to ignore the timestamp exposed by the exporter.

This feature is already implemented in prometheus. (cf. prometheus/prometheus#5302, prometheus/prometheus#5304)

### Current behavior:
Use the timestamp exposed by the exporter. 

### Desired behavior:
If `ignore_timestamp` is `false`, same as current behavior. (default)
If `ignore_timestamp` is `true`, use current timestamp.

### Use case: 
See these comments.
https://github.com/prometheus/prometheus/issues/5302#issuecomment-469641160
https://github.com/prometheus/prometheus/issues/5302#issuecomment-469642430

### PR DESCRIPTION
**Pull request by ykawamot**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. feat: or fix:)

resolves #9739

### CODE REVIEW

**Review by srebhan — COMMENTED** [review]
How about using the inverse (e.g. `ignore_timestamps`) with a default to `false` as this will introduce much less changes (as `false` is the golang default `nil` for boolean)?

**Comment by ykawamot** [comment]
@srebhan I thought so too, but I focused on making it work as well as `honor_timestamps`. (I'm not an enthusiast of prometheus, so there's no need to make it the same as its behavior.)
I think it is more reasonable to have the same default values as the other telegraf settings, so I will fix it.

File: plugins/inputs/prometheus/README.md [hunk 1]
```diff
@@ -23,6 +23,9 @@ in Prometheus format.
   ## Url tag name (tag containing scrapped url. optional, default is "url")
   # url_tag = "url"
   
+  ## Whether the timestamp of the scraped metrics will be ignored.
```
Comment (srebhan) [inline plugins/inputs/prometheus/README.md:26]:
```suggestion
  ## Whether the timestamp of the scraped metrics will be ignored.
  ## If set to true, the gather time will be used.
```

**Review by srebhan — COMMENTED** [review]
Hey @ykawamot, thanks for the modifications. I have only one small suggestion to improve the documentation. Can you please take a look?

**Comment by ykawamot** [comment]
@srebhan Thanks for your suggestion. I've modified the other two files as well.

**Pull request merged**